# Pathway enrichment

* Using `msigdbr` package in R to download GSEA pathway gene sets
* initially download Hallmark gene sets

In [1]:
#set wd
getwd()
setwd('/scratch/user/s4436039/scdata/Pathway-sets')
getwd()

[1] "/scratch/user/s4436039/Nikita-project"

[1] "/scratch/user/s4436039/scdata/Pathway-sets"

In [2]:
library(msigdbr)

In [3]:
msigdbr_collections()

gs_cat gs_subcat       num_genesets
1  C1                      299        
2  C2     CGP             3384        
3  C2     CP                29        
4  C2     CP:BIOCARTA      292        
5  C2     CP:KEGG          186        
6  C2     CP:PID           196        
7  C2     CP:REACTOME     1615        
8  C2     CP:WIKIPATHWAYS  664        
9  C3     MIR:MIRDB       2377        
10 C3     MIR:MIR_Legacy   221        
11 C3     TFT:GTRD         518        
12 C3     TFT:TFT_Legacy   610        
13 C4     CGN              427        
14 C4     CM               431        
15 C5     GO:BP           7658        
16 C5     GO:CC           1006        
17 C5     GO:MF           1738        
18 C5     HPO             5071        
19 C6                      189        
20 C7     IMMUNESIGDB     4872        
21 C7     VAX              347        
22 C8                      700        
23 H                        50

In [4]:
msigdbr_species()

species_name                   
1  Anolis carolinensis            
2  Bos taurus                     
3  Caenorhabditis elegans         
4  Canis lupus familiaris         
5  Danio rerio                    
6  Drosophila melanogaster        
7  Equus caballus                 
8  Felis catus                    
9  Gallus gallus                  
10 Homo sapiens                   
11 Macaca mulatta                 
12 Monodelphis domestica          
13 Mus musculus                   
14 Ornithorhynchus anatinus       
15 Pan troglodytes                
16 Rattus norvegicus              
17 Saccharomyces cerevisiae       
18 Schizosaccharomyces pombe 972h-
19 Sus scrofa                     
20 Xenopus tropicalis             
   species_common_name                                                    
1  Carolina anole, green anole                                            
2  bovine, cattle, cow, dairy cow, domestic cattle, domestic cow, ox, oxen
3  NA                                                                     
4  dog, dogs                                                              
5  leopard danio, zebra danio, zebra fish, zebrafish                      
6  fruit fly                                                              
7  domestic horse, equine, horse                                          
8  cat, cats, domestic cat                                                
9  bantam, chicken, chickens, Gallus domesticus                           
10 human                                                                  
11 rhesus macaque, rhesus macaques, Rhesus monkey, rhesus monkeys         
12 gray short-tailed opossum                                              
13 house mouse, mouse                                                     
14 duck-billed platypus, duckbill platypus, platypus                      
15 chimpanzee                                                             
16 brown rat, Norway rat, rat, rats                                       
17 baker's yeast, brewer's yeast, S. cerevisiae                           
18 NA                                                                     
19 pig, pigs, swine, wild boar                                            
20 tropical clawed frog, western clawed frog

In [5]:
#download human hallmark gene sets
h_gene_sets = msigdbr(species = "human", category = "H")
head(h_gene_sets)

gs_cat gs_subcat gs_name               gene_symbol entrez_gene
1 H                HALLMARK_ADIPOGENESIS ABCA1          19      
2 H                HALLMARK_ADIPOGENESIS ABCB8       11194      
3 H                HALLMARK_ADIPOGENESIS ACAA2       10449      
4 H                HALLMARK_ADIPOGENESIS ACADL          33      
5 H                HALLMARK_ADIPOGENESIS ACADM          34      
6 H                HALLMARK_ADIPOGENESIS ACADS          35      
  ensembl_gene    human_gene_symbol human_entrez_gene human_ensembl_gene gs_id
1 ENSG00000165029 ABCA1                19             ENSG00000165029    M5905
2 ENSG00000197150 ABCB8             11194             ENSG00000197150    M5905
3 ENSG00000167315 ACAA2             10449             ENSG00000167315    M5905
4 ENSG00000115361 ACADL                33             ENSG00000115361    M5905
5 ENSG00000117054 ACADM                34             ENSG00000117054    M5905
6 ENSG00000122971 ACADS                35             ENSG00000122971    M5905
  gs_pmid  gs_geoid gs_exact_source gs_url
1 26771021                                
2 26771021                                
3 26771021                                
4 26771021                                
5 26771021                                
6 26771021                                
  gs_description                                                     
1 Genes up-regulated during adipocyte differentiation (adipogenesis).
2 Genes up-regulated during adipocyte differentiation (adipogenesis).
3 Genes up-regulated during adipocyte differentiation (adipogenesis).
4 Genes up-regulated during adipocyte differentiation (adipogenesis).
5 Genes up-regulated during adipocyte differentiation (adipogenesis).
6 Genes up-regulated during adipocyte differentiation (adipogenesis).

In [7]:
unique(h_gene_sets$gs_name)

[1] "HALLMARK_ADIPOGENESIS"                     
 [2] "HALLMARK_ALLOGRAFT_REJECTION"              
 [3] "HALLMARK_ANDROGEN_RESPONSE"                
 [4] "HALLMARK_ANGIOGENESIS"                     
 [5] "HALLMARK_APICAL_JUNCTION"                  
 [6] "HALLMARK_APICAL_SURFACE"                   
 [7] "HALLMARK_APOPTOSIS"                        
 [8] "HALLMARK_BILE_ACID_METABOLISM"             
 [9] "HALLMARK_CHOLESTEROL_HOMEOSTASIS"          
[10] "HALLMARK_COAGULATION"                      
[11] "HALLMARK_COMPLEMENT"                       
[12] "HALLMARK_DNA_REPAIR"                       
[13] "HALLMARK_E2F_TARGETS"                      
[14] "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"
[15] "HALLMARK_ESTROGEN_RESPONSE_EARLY"          
[16] "HALLMARK_ESTROGEN_RESPONSE_LATE"           
[17] "HALLMARK_FATTY_ACID_METABOLISM"            
[18] "HALLMARK_G2M_CHECKPOINT"                   
[19] "HALLMARK_GLYCOLYSIS"                       
[20] "HALLMARK_HEDGEHOG_SIGNALING"               
[21] "HALLMARK_HEME_METABOLISM"                  
[22] "HALLMARK_HYPOXIA"                          
[23] "HALLMARK_IL2_STAT5_SIGNALING"              
[24] "HALLMARK_IL6_JAK_STAT3_SIGNALING"          
[25] "HALLMARK_INFLAMMATORY_RESPONSE"            
[26] "HALLMARK_INTERFERON_ALPHA_RESPONSE"        
[27] "HALLMARK_INTERFERON_GAMMA_RESPONSE"        
[28] "HALLMARK_KRAS_SIGNALING_DN"                
[29] "HALLMARK_KRAS_SIGNALING_UP"                
[30] "HALLMARK_MITOTIC_SPINDLE"                  
[31] "HALLMARK_MTORC1_SIGNALING"                 
[32] "HALLMARK_MYC_TARGETS_V1"                   
[33] "HALLMARK_MYC_TARGETS_V2"                   
[34] "HALLMARK_MYOGENESIS"                       
[35] "HALLMARK_NOTCH_SIGNALING"                  
[36] "HALLMARK_OXIDATIVE_PHOSPHORYLATION"        
[37] "HALLMARK_P53_PATHWAY"                      
[38] "HALLMARK_PANCREAS_BETA_CELLS"              
[39] "HALLMARK_PEROXISOME"                       
[40] "HALLMARK_PI3K_AKT_MTOR_SIGNALING"          
[41] "HALLMARK_PROTEIN_SECRETION"                
[42] "HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY"  
[43] "HALLMARK_SPERMATOGENESIS"                  
[44] "HALLMARK_TGF_BETA_SIGNALING"               
[45] "HALLMARK_TNFA_SIGNALING_VIA_NFKB"          
[46] "HALLMARK_UNFOLDED_PROTEIN_RESPONSE"        
[47] "HALLMARK_UV_RESPONSE_DN"                   
[48] "HALLMARK_UV_RESPONSE_UP"                   
[49] "HALLMARK_WNT_BETA_CATENIN_SIGNALING"       
[50] "HALLMARK_XENOBIOTIC_METABOLISM"

In [9]:
#seperate table based on gs_name (dataframe becomes a list of smaller dataframes based on gs_name)
h_sets_list <- split(h_gene_sets, h_gene_sets$gs_name)
head(h_sets_list$HALLMARK_ADIPOGENESIS)

gs_cat gs_subcat gs_name               gene_symbol entrez_gene
1 H                HALLMARK_ADIPOGENESIS ABCA1          19      
2 H                HALLMARK_ADIPOGENESIS ABCB8       11194      
3 H                HALLMARK_ADIPOGENESIS ACAA2       10449      
4 H                HALLMARK_ADIPOGENESIS ACADL          33      
5 H                HALLMARK_ADIPOGENESIS ACADM          34      
6 H                HALLMARK_ADIPOGENESIS ACADS          35      
  ensembl_gene    human_gene_symbol human_entrez_gene human_ensembl_gene gs_id
1 ENSG00000165029 ABCA1                19             ENSG00000165029    M5905
2 ENSG00000197150 ABCB8             11194             ENSG00000197150    M5905
3 ENSG00000167315 ACAA2             10449             ENSG00000167315    M5905
4 ENSG00000115361 ACADL                33             ENSG00000115361    M5905
5 ENSG00000117054 ACADM                34             ENSG00000117054    M5905
6 ENSG00000122971 ACADS                35             ENSG00000122971    M5905
  gs_pmid  gs_geoid gs_exact_source gs_url
1 26771021                                
2 26771021                                
3 26771021                                
4 26771021                                
5 26771021                                
6 26771021                                
  gs_description                                                     
1 Genes up-regulated during adipocyte differentiation (adipogenesis).
2 Genes up-regulated during adipocyte differentiation (adipogenesis).
3 Genes up-regulated during adipocyte differentiation (adipogenesis).
4 Genes up-regulated during adipocyte differentiation (adipogenesis).
5 Genes up-regulated during adipocyte differentiation (adipogenesis).
6 Genes up-regulated during adipocyte differentiation (adipogenesis).

In [ ]:
#save as a excel file (each sheet will be a seperate gene set/ dataframe)
library(openxlsx)

In [ ]:
# Assign numeric sheet names (keeping it as the gene_set names made the sheet names too long)
numeric_names <- paste0("Sheet", seq_along(h_sets_list))

In [ ]:
# Write to Excel
write.xlsx(h_sets_list, file = "hallmark_sets.xlsx", sheetName = numeric_names)

### Next downloading select KEGG pathways:
* all C2 --> CP --> CP:KEGG_MEDICUS
* KEGG_MEDICUS_REFERENCE_ANTIGEN_PROCESSING_AND_PRESENTATION_BY_MHC_CLASS_I_MOLECULES
* KEGG_MEDICUS_REFERENCE_ANTIGEN_PROCESSING_AND_PRESENTATION_BY_MHC_CLASS_II_MOLECULES
* KEGG_MEDICUS_REFERENCE_CD80_CD86_CD28_PI3K_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_CD80_CD86_CTLA4_PP2A_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_CGAS_STING_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_CHOLESTEROL_BIOSYNTHESIS
* KEGG_MEDICUS_REFERENCE_CROSSTALK_BETWEEN_EXTRINSIC_AND_INTRINSIC_APOPTOTIC_PATHWAYS
* KEGG_MEDICUS_REFERENCE_CYTOKINE_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_EARLY_ENDOSOMAL_FUSION
* KEGG_MEDICUS_REFERENCE_EGF_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_EXTRINSIC_APOPTOTIC_PATHWAY
* KEGG_MEDICUS_REFERENCE_GLYCOLYSIS
* KEGG_MEDICUS_REFERENCE_IL10_FAMILY_TO_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_IL12_23_TO_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_IL1_IL1R_JNK_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_IL1_IL1R_P38_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_IL2_FAMILY_TO_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_IL6_FAMILY_TO_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_INTRINSIC_APOPTOTIC_PATHWAY
* KEGG_MEDICUS_REFERENCE_LEWIS_X_ANTIGEN_BIOSYNTHESIS
* KEGG_MEDICUS_REFERENCE_LYSOSOMAL_CA2_RELEASE
* KEGG_MEDICUS_REFERENCE_MISMATCH_REPAIR
* KEGG_MEDICUS_REFERENCE_NOTCH_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TGF_BETA_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR1_2_4_NFKB_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR2_4_MAPK_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR3_IRF3_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR3_IRF7_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR3_NFKB_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR4_IRF3_7_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR5_NFKB_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR7_8_9_IRF5_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TLR7_9_IRF7_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TNF_JNK_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TNF_NFKB_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TYPE_I_IFN_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TYPE_I_INTERFERON_TO_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MEDICUS_REFERENCE_TYPE_II_INTERFERON_TO_JAK_STAT_IGNALING_PATHWAY

### found out msigdbr just has Kegg legacy sets (186 sets)
* all C2 --> CP --> CP:KEGG_LEGACY
* KEGG_ADIPOCYTOKINE_SIGNALING_PATHWAY
* KEGG_ANTIGEN_PROCESSING_AND_PRESENTATION
* KEGG_APOPTOSIS
* KEGG_CALCIUM_SIGNALING_PATHWAY
* KEGG_CELL_ADHESION_MOLECULES_CAMS
* KEGG_CHEMOKINE_SIGNALING_PATHWAY
* KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION
* KEGG_CYTOSOLIC_DNA_SENSING_PATHWAY
* KEGG_ENDOCYTOSIS
* KEGG_FATTY_ACID_METABOLISM
* KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS
* KEGG_GLYCOLYSIS_GLUCONEOGENESIS
* KEGG_JAK_STAT_SIGNALING_PATHWAY
* KEGG_MAPK_SIGNALING_PATHWAY
* KEGG_NOD_LIKE_RECEPTOR_SIGNALING_PATHWAY
* KEGG_NOTCH_SIGNALING_PATHWAY
* KEGG_OXIDATIVE_PHOSPHORYLATION
* KEGG_PROTEASOME
* KEGG_PROTEIN_EXPORT
* KEGG_TGF_BETA_SIGNALING_PATHWAY
* KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY
* KEGG_TYROSINE_METABOLISM
* KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS
* KEGG_VEGF_SIGNALING_PATHWAY


In [4]:
msigdbr_collections()

gs_cat gs_subcat       num_genesets
1  C1                      299        
2  C2     CGP             3384        
3  C2     CP                29        
4  C2     CP:BIOCARTA      292        
5  C2     CP:KEGG          186        
6  C2     CP:PID           196        
7  C2     CP:REACTOME     1615        
8  C2     CP:WIKIPATHWAYS  664        
9  C3     MIR:MIRDB       2377        
10 C3     MIR:MIR_Legacy   221        
11 C3     TFT:GTRD         518        
12 C3     TFT:TFT_Legacy   610        
13 C4     CGN              427        
14 C4     CM               431        
15 C5     GO:BP           7658        
16 C5     GO:CC           1006        
17 C5     GO:MF           1738        
18 C5     HPO             5071        
19 C6                      189        
20 C7     IMMUNESIGDB     4872        
21 C7     VAX              347        
22 C8                      700        
23 H                        50

In [5]:
# download human KEGG gene sets
kegg_sets = msigdbr(species = "human", category = "C2", subcategory = "CP:KEGG")
head(kegg_sets)

gs_cat gs_subcat gs_name               gene_symbol entrez_gene
1 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA1           19     
2 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA10       10349     
3 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA12       26154     
4 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA13      154664     
5 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA2           20     
6 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA3           21     
  ensembl_gene    human_gene_symbol human_entrez_gene human_ensembl_gene gs_id 
1 ENSG00000165029 ABCA1                 19            ENSG00000165029    M11911
2 ENSG00000154263 ABCA10             10349            ENSG00000154263    M11911
3 ENSG00000144452 ABCA12             26154            ENSG00000144452    M11911
4 ENSG00000179869 ABCA13            154664            ENSG00000179869    M11911
5 ENSG00000107331 ABCA2                 20            ENSG00000107331    M11911
6 ENSG00000167972 ABCA3                 21            ENSG00000167972    M11911
  gs_pmid gs_geoid gs_exact_source
1                  hsa02010       
2                  hsa02010       
3                  hsa02010       
4                  hsa02010       
5                  hsa02010       
6                  hsa02010       
  gs_url                                              gs_description  
1 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
2 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
3 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
4 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
5 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
6 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters

In [6]:
unique(kegg_sets$gs_name)

[1] "KEGG_ABC_TRANSPORTERS"                                          
  [2] "KEGG_ACUTE_MYELOID_LEUKEMIA"                                    
  [3] "KEGG_ADHERENS_JUNCTION"                                         
  [4] "KEGG_ADIPOCYTOKINE_SIGNALING_PATHWAY"                           
  [5] "KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM"                
  [6] "KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION"                 
  [7] "KEGG_ALLOGRAFT_REJECTION"                                       
  [8] "KEGG_ALPHA_LINOLENIC_ACID_METABOLISM"                           
  [9] "KEGG_ALZHEIMERS_DISEASE"                                        
 [10] "KEGG_AMINOACYL_TRNA_BIOSYNTHESIS"                               
 [11] "KEGG_AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM"               
 [12] "KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS"                         
 [13] "KEGG_ANTIGEN_PROCESSING_AND_PRESENTATION"                       
 [14] "KEGG_APOPTOSIS"                                                 
 [15] "KEGG_ARACHIDONIC_ACID_METABOLISM"                               
 [16] "KEGG_ARGININE_AND_PROLINE_METABOLISM"                           
 [17] "KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC"      
 [18] "KEGG_ASCORBATE_AND_ALDARATE_METABOLISM"                         
 [19] "KEGG_ASTHMA"                                                    
 [20] "KEGG_AUTOIMMUNE_THYROID_DISEASE"                                
 [21] "KEGG_AXON_GUIDANCE"                                             
 [22] "KEGG_BASAL_CELL_CARCINOMA"                                      
 [23] "KEGG_BASAL_TRANSCRIPTION_FACTORS"                               
 [24] "KEGG_BASE_EXCISION_REPAIR"                                      
 [25] "KEGG_BETA_ALANINE_METABOLISM"                                   
 [26] "KEGG_BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS"                   
 [27] "KEGG_BLADDER_CANCER"                                            
 [28] "KEGG_BUTANOATE_METABOLISM"                                      
 [29] "KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY"                         
 [30] "KEGG_CALCIUM_SIGNALING_PATHWAY"                                 
 [31] "KEGG_CARDIAC_MUSCLE_CONTRACTION"                                
 [32] "KEGG_CELL_ADHESION_MOLECULES_CAMS"                              
 [33] "KEGG_CELL_CYCLE"                                                
 [34] "KEGG_CHEMOKINE_SIGNALING_PATHWAY"                               
 [35] "KEGG_CHRONIC_MYELOID_LEUKEMIA"                                  
 [36] "KEGG_CIRCADIAN_RHYTHM_MAMMAL"                                   
 [37] "KEGG_CITRATE_CYCLE_TCA_CYCLE"                                   
 [38] "KEGG_COLORECTAL_CANCER"                                         
 [39] "KEGG_COMPLEMENT_AND_COAGULATION_CASCADES"                       
 [40] "KEGG_CYSTEINE_AND_METHIONINE_METABOLISM"                        
 [41] "KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION"                    
 [42] "KEGG_CYTOSOLIC_DNA_SENSING_PATHWAY"                             
 [43] "KEGG_DILATED_CARDIOMYOPATHY"                                    
 [44] "KEGG_DNA_REPLICATION"                                           
 [45] "KEGG_DORSO_VENTRAL_AXIS_FORMATION"                              
 [46] "KEGG_DRUG_METABOLISM_CYTOCHROME_P450"                           
 [47] "KEGG_DRUG_METABOLISM_OTHER_ENZYMES"                             
 [48] "KEGG_ECM_RECEPTOR_INTERACTION"                                  
 [49] "KEGG_ENDOCYTOSIS"                                               
 [50] "KEGG_ENDOMETRIAL_CANCER"                                        
 [51] "KEGG_EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION"
 [52] "KEGG_ERBB_SIGNALING_PATHWAY"                                    
 [53] "KEGG_ETHER_LIPID_METABOLISM"                                    
 [54] "KEGG_FATTY_ACID_METABOLISM"                                     
 [55] "KEGG_FC_EPSILON_RI_SIGNALING_PATHWAY"                           
 [56] "KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOS

In [7]:
#seperate table based on gs_name (dataframe becomes a list of smaller dataframes based on gs_name)
kegg_sets_list <- split(kegg_sets, kegg_sets$gs_name)
head(kegg_sets_list$KEGG_ABC_TRANSPORTERS)

gs_cat gs_subcat gs_name               gene_symbol entrez_gene
1 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA1           19     
2 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA10       10349     
3 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA12       26154     
4 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA13      154664     
5 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA2           20     
6 C2     CP:KEGG   KEGG_ABC_TRANSPORTERS ABCA3           21     
  ensembl_gene    human_gene_symbol human_entrez_gene human_ensembl_gene gs_id 
1 ENSG00000165029 ABCA1                 19            ENSG00000165029    M11911
2 ENSG00000154263 ABCA10             10349            ENSG00000154263    M11911
3 ENSG00000144452 ABCA12             26154            ENSG00000144452    M11911
4 ENSG00000179869 ABCA13            154664            ENSG00000179869    M11911
5 ENSG00000107331 ABCA2                 20            ENSG00000107331    M11911
6 ENSG00000167972 ABCA3                 21            ENSG00000167972    M11911
  gs_pmid gs_geoid gs_exact_source
1                  hsa02010       
2                  hsa02010       
3                  hsa02010       
4                  hsa02010       
5                  hsa02010       
6                  hsa02010       
  gs_url                                              gs_description  
1 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
2 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
3 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
4 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
5 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters
6 http://www.genome.jp/kegg/pathway/hsa/hsa02010.html ABC transporters

In [8]:
#save as a excel file (each sheet will be a seperate gene set/ dataframe)
library(openxlsx)

In [9]:
# Assign numeric sheet names (keeping it as the gene_set names made the sheet names too long)
numeric_names <- paste0("Sheet", seq_along(kegg_sets_list))

In [10]:
# Write to Excel
write.xlsx(kegg_sets_list, file = "kegg_sets.xlsx", sheetName = numeric_names)